app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialFragment.java

```diff
@@ -363,8 +402,123 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     ready.onNext(null);
   }
 
-  @Override public Observable<String> mediaContentClicked() {
-    return editorialMediaClicked;
+  @Override public Observable<ScrollEvent> placeHolderVisibilityChange() {
+    return RxNestedScrollView.scrollChangeEvents(scrollView)
+        .flatMap(viewScrollChangeEvent -> Observable.just(viewScrollChangeEvent)
+            .map(scrollDown -> isItemShown())
+            .distinctUntilChanged()
+            .map(isItemShown -> new ScrollEvent(
+                isScrollDown(viewScrollChangeEvent.oldScrollY(), viewScrollChangeEvent.scrollY()),
+                isItemShown)));
+  }
+
+  @Override public void removeBottomCardAnimation() {
+    View view = getPlaceHolder();
+    if (!bottomCardHidden && view != null) {
+      appCardLayout.animate()
+          .scaleY(0f)
+          .scaleX(0f)
+          .alpha(0)
+          .setDuration(300)
+          .setListener(new Animator.AnimatorListener() {
+            @Override public void onAnimationStart(Animator animator) {
+              View view = adapter.getPlaceHolder();
+              view.animate()
+                  .scaleX(1f)
+                  .scaleY(1f)
+                  .alpha(1)
+                  .setDuration(300)
+                  .setListener(new Animator.AnimatorListener() {
+                    @Override public void onAnimationStart(Animator animator) {
+                      view.setVisibility(View.VISIBLE);
+                    }
+
+                    @Override public void onAnimationEnd(Animator animator) {
+
+                    }
+
+                    @Override public void onAnimationCancel(Animator animator) {
+
+                    }
+
+                    @Override public void onAnimationRepeat(Animator animator) {
+
+                    }
+                  })
+                  .start();
+            }
+
+            @Override public void onAnimationEnd(Animator animator) {
+              bottomCardHidden = true;
+            }
+
+            @Override public void onAnimationCancel(Animator animator) {
+
+            }
+
+            @Override public void onAnimationRepeat(Animator animator) {
+
+            }
+          })
+          .start();
+    }
+  }
+
+  @Override public void addBottomCardAnimation() {
+    View view = getPlaceHolder();
+    if (bottomCardHidden && view != null) {
+      view.animate()
```

Comment:
same as above, i would recommend extracting this configuration to a method just to improve code readibility

Reply:
Fixed